include/seqan3/std/charconv

```diff
@@ -13,97 +13,30 @@
- * | gcc 7          | undefined and NO `<charconv>` header | undefined                 | contrib (llvm) | contrib (llvm)  | contrib (llvm)    | contrib (llvm) | contrib (llvm)   | shim (ostringstream) | shim (strto[f/d/ld]) |
- * | gcc 8          | undefined and `<charconv>` header    | 100000                    | contrib (llvm) | stdlib          | stdlib            | stdlib         | stdlib           | shim (ostringstream) | shim (strto[f/d/ld]) |
- * | gcc 9          | undefined and `<charconv>` header    | 100000                    | contrib (llvm) | stdlib          | stdlib            | stdlib         | stdlib           | shim (ostringstream) | shim (strto[f/d/ld]) |
- * | gcc 10         | undefined and `<charconv>` header    | 100000                    | stdlib         | stdlib          | stdlib            | stdlib         | stdlib           | shim (ostringstream) | shim (strto[f/d/ld]) |
- * | gcc 11         | undefined and `<charconv>` header    | 100000* (could be 201611) | stdlib         | stdlib          | stdlib            | stdlib         | stdlib           | stdlib               | stdlib               |
+ * | stdlib version | __cpp_lib_to_chars                               | chars_format   | to_chars_result | from_chars_result | to_chars (int) | from_chars (int) | to_chars (float)     | from_chars (float)   |
+ * | -------------- | ------------------------------------------------ | -------------- | --------------- | ----------------- | -------------- | ---------------- | -------------------- | -------------------- |
+ * | gcc 10         | undefined and `<charconv>` header                | stdlib         | stdlib          | stdlib            | stdlib         | stdlib           | shim (ostringstream) | shim (strto[f/d/ld]) |
+ * | gcc 11         | undefined (or 201611) and `<charconv>` header    | stdlib         | stdlib          | stdlib            | stdlib         | stdlib           | stdlib               | stdlib               |
  *
  * Note: gcc 11 implements float too, but does not define __cpp_lib_to_chars
  */
-#ifdef SEQAN3_DOXYGEN_ONLY // needed as seqan3/core/platform.hpp might not be included
-#if SEQAN3_DOXYGEN_ONLY(1)0
-// documentation-only
-#define SEQAN3_CPP_LIB_TO_CHARS IMPLEMENTATION_DEFINED
-#endif // SEQAN3_DOXYGEN_ONLY(1)0
-#endif // SEQAN3_DOXYGEN_ONLY
-
-#if __cpp_lib_to_chars >= 201611 // iff complete (int AND float) charconv implementation
-#  define SEQAN3_CPP_LIB_TO_CHARS __cpp_lib_to_chars
-#elif __has_include(<charconv>) // iff incomplete (only int) charconv implementation
-#  define SEQAN3_CPP_LIB_TO_CHARS 100000
-#endif // __cpp_lib_to_chars >= 201611
 
 // =========================================================================
-// If any std implementation is present use that as basis (>= gcc8)
+// If float implementation is missing, add our own shim-implementation
```

Comment:
I think this should still be `any std`?

Reply:
`int` is provided for gcc >= 10, `float` for >= 11, I think.
I removed the int shim, so we only ever add float